### PR TITLE
League training with slime volley env

### PR DESCRIPTION
**Pull request by LuciusMos**
1. Add slimevolleygym env in dizoo.
2. League training pipeline (Not naive self-play, but league training)
3. PPO tuning

### CODE REVIEW

File: dizoo/slime_volley/envs/test_slam_volley_env.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import pytest
+import numpy as np
+from easydict import EasyDict
+
+from dizoo.slime_volley.envs.slime_volley_env import SlimeVolleyEnv
+
+
+@pytest.mark.unittest
+class TestOvercooked:
```
Comment (zxzzz0) [inline dizoo/slime_volley/envs/test_slam_volley_env.py:9]: This name, `TestOvercooked`, might be changed.

File: dizoo/slime_volley/entry/slime_volley_league_ppo_main.py [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+        return {env_id: {'action': torch.zeros(1)} for env_id in data.keys()}
+
+    def reset(self, data_id: list = []) -> None:
+        pass
+
+
+class EvalPolicy2:
+
+    def forward(self, data: dict) -> dict:
+        return {env_id: {'action': torch.from_numpy(np.random.randint(0, 6, (1, )))} for env_id in data.keys()}
```
Comment (zxzzz0) [inline dizoo/slime_volley/entry/slime_volley_league_ppo_main.py:32]: > evaluator is not finished, now use a very naive one
Actually there is a built-in AI that you can use for evaluation.

**Comment by zxzzz0** [comment]
@PaParaZz1 Could we fix the unit test failure?

File: dizoo/slime_volley/config/slime_volley_league_ppo_config.py [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+                    strong_win_rate=0.7,
+                    min_valid_win_rate=0.3,
+                ),
+                league_exploiter=dict(
+                    one_phase_step=200,
+                    branch_probs=dict(pfsp=1.0, ),
+                    strong_win_rate=0.7,
+                    mutate_prob=0.0,
+                ),
+                use_pretrain=False,
```
Comment (zxzzz0) [inline dizoo/slime_volley/config/slime_volley_league_ppo_config.py:67]: @PaParaZz1  Follow up on [this](https://github.com/opendilab/DI-engine/discussions/61#discussioncomment-1412106). Is it possible that we can have another new config like `xxx_league_PPO_using_pretrain_config.py`, where `use_pretrain` is set to `True` rather than `False` here so that we don't train from scratch (Motivation is that training from scratch can be **_slow_** and we believe using pretrain model can benefit the training)

 In this new config each time the model is mutated, it will be reset to the pretrain model. Assume you obtained this pretrain model offline by running PPO to train with the built-in bot `dizoo/slime_volley/config/slime_volley_ppo_config.py` for a very short period of time.

**Comment by PaParaZz1** [comment]
Here are `DI-engine PPO` (vs bot) results in `SlimeVolley-v0` env (5 random seed):

Screen Shot 2021-10-05 at 10 12 06 PM

Screen Shot 2021-10-05 at 10 12 26 PM

**Comment by zxzzz0** [comment]
> Here are `DI-engine PPO` (vs bot) results in `SlimeVolley-v0` env (5 random seed):

@PaParaZz1  Well done! What a nice curve. This curve is very similar to the other people's [PPO study](https://github.com/hardmaru/slimevolleygym/raw/master/figure/mpi_ppo_results.svg) as reported [here](https://github.com/hardmaru/slimevolleygym/blob/master/TRAINING.md#ppo-and-cma-es-example-train-directly-against-baseline-policy).
